plugins/kubernetes/app/controllers/kubernetes/deploy_group_roles_controller.rb

```diff
@@ -28,9 +30,40 @@ def index
     @deploy_group_roles = @deploy_group_roles.
       joins(:project, :kubernetes_role).
       order('projects.name, kubernetes_roles.name')
+
+    respond_to do |format|
+      format.html
+      format.json { render json: {deploy_group_roles: @deploy_group_roles} }
+    end
   end
 
   def show
+    respond_to do |format|
+      format.html
+      format.json do
+        deploy_group_role = @deploy_group_role.as_json
+        if params[:include].present? && params[:include].to_s.split(',').include?("template")
+
+          release = Kubernetes::Release.create(
+            git_sha: params['git_sha'] || DEFAULT_BRANCH,
+            git_ref: params['git_ref'] || DEFAULT_BRANCH,
+            project: Project.find_by_id(@deploy_group_role.project_id),
```

Comment:
this looks dangerous ... if I set a specific sha then the ref should not get set to master ?

Reply:
`.find` is fine ... blow up when there is no project

Reply:
blow up how? what do i raise? 

didn't find any good example of errors/messages for json responses..

If i just give a bogus id locally, i get an `ActiveRecord::NotFound` exception, but no idea if that is wrapped by something if it's configured for production?